src/cf.data.pre

```diff
@@ -2379,6 +2379,18 @@ DOC_START
 			certificates. If set to zero, caching is disabled. The
 			default value is 4MB.
 
+	    ssl-bump also accepts the option:
+	    
+	   terminate-on-secure-connect-fail
+			Workaround for incompatibility with HTTPS Upgrade (Automatic HTTPS)
+			introduced in Chromium browsers.
+
+			With this option set, if CONNECT (including certificate validation) to
+			a site failed, Squid will terminate the client session without sending
+			any header and body.
+			This enables the affected browsers to fall back to HTTP instead of
+			showing the error page generated by Squid.
+
```

Comment:
My explanation in this PR might have an unclear point, so I leave a note first here just in case:

Squid returns "HTTP/1.1 200" for CONNECT to affected sites even if this draft option is set.

After receiving the actual request (the first HTTP request as you mention) from the client, Squid without the option returns the 503 error page (ERR_SECURE_CONNECT_FAIL/ERR_CONNECT_FAIL), and Squid with the option set returns nothing (terminates TCP with no more header/body).

The browser shows ERR_EMPTY_RESPONSE message for the latter case (if intentionally tried to access the URL of wrong HTTPS, not by HTTPS Upgrade).

Reply:
FWIW, you comment matches my overall understanding of this PR. The only difference I am aware of is that Squid does not close the connection to the browser in my tests and "terminates TCP" in yours. We are discussing that aspect in [another change request](https://github.com/squid-cache/squid/pull/1668#discussion_r1486655553).